### PR TITLE
fix: fixed the error we see while running the package(s) in dry-mode

### PR DESCRIPTION
**Pull request by Peeeekay**
This PR fixes this issue:- https://app.zenhub.com/workspaces/engineering-636cff9fc978ceb2aac05a1d/issues/gh/kurtosis-tech/kurtosis/657

### CODE REVIEW

**Review by h4ck3rk3y — APPROVED** [review]
LGTM

File: core/server/api_container/server/startosis_engine/startosis_executor.go [hunk 1]
```diff
@@ -75,15 +75,17 @@ func (executor *StartosisExecutor) Execute(ctx context.Context, dryRun bool, par
...
 
-		scriptWithValuesReplaced, err := magic_string_helper.ReplaceRuntimeValueInString(serializedScriptOutput, executor.runtimeValueStore)
-		if err != nil {
-			propagatedErr := stacktrace.Propagate(err, "An error occurred while replacing the runtime values in the output of the script")
-			serializedError := binding_constructors.NewStarlarkExecutionError(propagatedErr.Error())
-			starlarkRunResponseLineStream <- binding_constructors.NewStarlarkRunResponseLineFromExecutionError(serializedError)
-			starlarkRunResponseLineStream <- binding_constructors.NewStarlarkRunResponseLineFromRunFailureEvent()
-			return
+		if !dryRun {
+			scriptWithValuesReplaced, err := magic_string_helper.ReplaceRuntimeValueInString(serializedScriptOutput, executor.runtimeValueStore)
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/startosis_executor.go:79]: why did dry run only fail for exec and not the others?

File: core/server/api_container/server/startosis_engine/startosis_executor.go [hunk 2]
```diff
@@ -75,15 +75,17 @@ func (executor *StartosisExecutor) Execute(ctx context.Context, dryRun bool, par
...
 		}
 
-		scriptWithValuesReplaced, err := magic_string_helper.ReplaceRuntimeValueInString(serializedScriptOutput, executor.runtimeValueStore)
-		if err != nil {
-			propagatedErr := stacktrace.Propagate(err, "An error occurred while replacing the runtime values in the output of the script")
-			serializedError := binding_constructors.NewStarlarkExecutionError(propagatedErr.Error())
-			starlarkRunResponseLineStream <- binding_constructors.NewStarlarkRunResponseLineFromExecutionError(serializedError)
-			starlarkRunResponseLineStream <- binding_constructors.NewStarlarkRunResponseLineFromRunFailureEvent()
-			return
+		if !dryRun {
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/startosis_executor.go:78]: can we add a --dry-run test in the CI for something that has runtime values?

Comment (Peeeekay) [reply core/server/api_container/server/startosis_engine/startosis_executor.go:78]: true; let me do that in some other PR. I will probably just use some existing package and add that in the CI.

**Pull request merged**